src/expr/node_algorithm.h

```diff
@@ -79,6 +79,10 @@ bool hasBoundVar(TNode n);
 /**
  * Returns true iff the node n contains a free variable, that is, a node
  * of kind BOUND_VARIABLE that is not bound in n.
+ *
+ * NOTE: this method should only be called on terms that do not have shadowing,
+ * which is the case if n is rewritten.
```

Comment:
clarify the behavior otherwise (like, does it crash or just does not fulfill the expected behavior)